source/extensions/path/pattern_template_lib/pattern_template.h

```diff
@@ -30,25 +30,28 @@ absl::StatusOr<std::string> convertURLPatternSyntaxToRegex(absl::string_view url
 // either literals or variable names).
 absl::StatusOr<std::vector<RewritePatternSegment>> parseRewritePattern(absl::string_view pattern);
 
-// Returns the parsed Url rewrite pattern to be used by
-// RewriteURLTemplatePattern()  |capture_regex| should
-// be the regex generated by ConvertURLPatternSyntaxToRegex().
+// Returns the parsed Url rewrite pattern and processes variables to be used by
+// RewriteURLTemplatePattern() and in rewrite.
+// |capture_regex| should be the regex generated by ConvertURLPatternSyntaxToRegex().
 absl::StatusOr<envoy::extensions::pattern_template::PatternTemplateRewriteSegments>
 parseRewritePattern(absl::string_view pattern, absl::string_view capture_regex);
 
 // Returns if provided rewrite pattern is valid
-absl::Status isValidPathTemplateRewritePattern(const std::string& path_template_rewrite);
+absl::Status isValidPathTemplateRewritePattern(absl::string_view path_template_rewrite);
 
-// Returns if path_template and rewrite_template have valid variables.
-// Every variable in rewrite MUST be present in match
+// Returns if path_template_rewrite and capture_regex have valid variables.
+// Every variable in rewrite MUST be present in match.
 // For example:
 // Match: /foo/{bar}/{var} and Rewrite: /goo/{var} is valid
 // Match: /foo/{bar} and Rewrite: /goo/{bar}/{var} is invalid. Match is missing {var}.
-absl::Status isValidSharedVariableSet(const std::string& path_template_rewrite,
-                                      const std::string& capture_regex);
+absl::Status isValidSharedVariableSet(absl::string_view path_template_rewrite,
+                                      absl::string_view capture_regex);
 
-absl::Status isValidMatchPattern(const std::string match_pattern);
+// Returns is the match_pattern is valid.
+// Validation attempts to parse pattern into literals and variables.
+absl::Status isValidMatchPattern(absl::string_view match_pattern);
 
+// Concatenates literals and extracts variable values to form the final rewritten url.
```

Comment:
Adding an example here would be helpful. Presumably, `url` is a raw URL which has not had any substitutions. Like "https://example.com/foo/bar". Then `capture_regex` is something like "/foo/{bar}", and rewrite_pattern is ...? i'm not quite sure, but and example would be helpful.